R/submodule.R

```diff
@@ -0,0 +1,111 @@
+parse_submodules <- function(file) {
+  if (grepl("\n", file)) {
+    x <- strsplit(file, "\n")[[1]]
+  } else {
+    x <- readLines(file)
+  }
+
+  # https://git-scm.com/docs/git-config#_syntax
+  # Subsection names are case sensitive and can contain any characters except
+  # newline and the null byte. Doublequote " and backslash can be included by
+  # escaping them as \" and \\
+  double_quoted_string_with_escapes <- '(?:\\\\.|[^"])*'
+
+  sections <- regexpr(
```

Comment:
I wonder if we should just include a collated rematch2 package, that would simplify things.